ballerina/tests/tests.bal

```diff
@@ -0,0 +1,592 @@
+        "label": updatedLabel
+    });
+
+    // Verify the patch
+    test:assertEquals(updatedLabel, response.label, "Pipeline label should be updated");
+    test:assertEquals(tempPipeline.stages.length(), response.stages.length(), "Stages should remain unchanged");
+    test:assertEquals(tempPipeline.displayOrder, response.displayOrder, "Display order should remain unchanged");
+
+    // Cleanup
+    _ = check hubspot->/[objectType]/[pipelineId].delete();
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+isolated function testDeletePipeline() returns error? {
+    // Create a test pipeline first
+    string objectType = "orders";
+    string uniqueLabel = string `Pipeline_${time:utcNow()[0]}`;
+    Pipeline response = check hubspot->/[objectType].post(payload = {
+        "displayOrder": 0,
+        "stages": [
+            {
+                "label": "In Progress",
+                "metadata": {
+                    "ticketState": "OPEN"
+                },
+                "displayOrder": 0
+            },
+            {
+                "label": "Done",
+                "metadata": {
+                    "ticketState": "CLOSED"
+                },
+                "displayOrder": 1
+            }
+        ],
+        "label": uniqueLabel
+    });
+
+    string pipelineId = response.id;
+
+    // Delete the pipeline
+    _ = check hubspot->/[objectType]/[pipelineId].delete();
+
+    // Verify deletion by attempting to get the pipeline
+    CollectionResponsePipelineNoPaging pipelines = check hubspot->/orders.get();
+    boolean pipelineExists = false;
+    foreach Pipeline pipeline in pipelines.results {
+        if pipeline.id == pipelineId {
+            pipelineExists = true;
+            break;
+        }
+    }
+
+    test:assertFalse(pipelineExists, "Pipeline should be deleted");
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+isolated function testGetPipelineAuditLog() returns error? {
+    string pipelineId = "673651319";
+
+    CollectionResponsePublicAuditInfoNoPaging|error response = hubspot->/["orders"]/[pipelineId]/audit.get();
+
+    if response is CollectionResponsePublicAuditInfoNoPaging {
+        test:assertTrue(response.results.length() > 0, "Pipeline audit should have at least one entry");
+    } else {
+        return response;
+    }
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+isolated function testGetPipelineStageAuditLog() returns error? {
+    string pipelineId = "673651319";
+    string stageId = "987806513"; // Replace with actual stage ID
+    string objectType = "orders";
+
+    CollectionResponsePublicAuditInfoNoPaging|error response =
+        check hubspot->/[objectType]/[pipelineId]/stages/[stageId]/audit.get();
+
+    if response is CollectionResponsePublicAuditInfoNoPaging {
+        test:assertTrue(response.results.length() > 0, "Pipeline stage audit should have at least one entry");
+    } else {
+        return response;
+    }
```

Comment:
You don't have to manually check the type, when the `check` expression is used. Check similar places.

```suggestion
    CollectionResponsePublicAuditInfoNoPaging response =
        check hubspot->/[objectType]/[pipelineId]/stages/[stageId]/audit.get();
    test:assertTrue(response.results.length() > 0, "Pipeline stage audit should have at least one entry");
```